### PR TITLE
Homework js2 week2

### PR DESCRIPTION
**Pull request by pallabi-ghosh**

### CODE REVIEW

**Review by flom84 — CHANGES_REQUESTED** [review]
For the movies exercise, please not include all the movies array...it's 40000 lines. is not okay...you should not make your own movies variable. Instead include javascript2/week2/homework/movies.js into your .html file

File: javascript/javascript2/week2/movie_function.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    return title.filter((filter) => {
+        const movieFilter = !filter.movies || listing.movies.includes('benjamin')
+        return movie
+    })
+}
+
+const filter = {
+    type: 'surfer',
+};
+console.log(moviesWithSpecialWords)
```
Comment (braedongough) [inline javascript/javascript2/week2/movie_function.js]: It doesn;t look like you're invoking the function here, that could be the cause of some of your problems: 
```suggestion
console.log(moviesWithSpecialWords())
```

File: javascript/javascript2/week2/movie_function.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+console.log(movieRatings.reduce(reducer));
+
+//Total Number of Movies containig Keywords (SURFER, ALIEN, BENJAMIN)
+
+const moviesWithSpecialWords = (title, filter) => {
+    return title.filter((filter) => {
+        const movieFilter = !filter.movies || listing.movies.includes('benjamin')
+        return movie
+    })
+}
```
Comment (braedongough) [inline javascript/javascript2/week2/movie_function.js]: As promise, here's how I would solve this one: 
```js
const numberOfMoviesWithKeyWords = (keywords) => {
  return movies.filter((movie) => {
    return keywords.some((keyword) => movie.title.includes(keyword))
  }).length
}

console.log(
  'numberOfMoviesWithKeyWords',
  numberOfMoviesWithKeyWords(['Alien', 'gai'])
)

```

The function I've created above allows you to pass any array of keywords to check against. I recommend you read into the `.some` array method. As you continue on you'll need to start reaching for some new tools to help with some more tricky problems  🤠 
 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

File: javascript/javascript2/week2/movie_function.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+const movies = [{ "title": "'71", "year": 2014, "rating": 7.2, "votes": 41702, "running_times": 5940 }, { "title": "'A' gai wak", "year": 1983, "rating": 7.4, "votes": 11942, "running_times": 6300 }, { "title": "'Breaker' Morant", "year": 1980, "rating": 7.9, "votes": 10702, "running_times": 6420 }, { "title": "'Crocodile' Dundee II", "year": 1988, "rating": 5.5, "votes": 47180, "running_times": 6480 }]
+
+
+function findShortest(movies) {
+    return movies.reduce(function(prevWord, currWord) {
```
Comment (braedongough) [inline javascript/javascript2/week2/movie_function.js:5]: Nice use of reduce! Reduce is a tricky method to learn so make sure to double check you've got the right solution here. It looks good from what I can see. See if you can refactor this to use an arrow function!

File: javascript/javascript2/week2/main.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+console.log('Script loaded');
+
+const products = getAvailableProducts();
+const productsUl = document.querySelector('section.products ul');
+console.log(productsUl);
```
Comment (braedongough) [inline javascript/javascript2/week2/main.js:5]: Remember to remove console logs that you only use for debugging from your final submission!

File: javascript/javascript2/week2/main.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+const input = document.querySelector('input');
+input.addEventListener('input', addtext);
+
+function addtext() {
+    let inputadd = document.querySelector('input');
```
Comment (braedongough) [inline javascript/javascript2/week2/main.js:11]: You've already selected this above, you can reuse the same variable `input`!

**Review by braedongough — APPROVED** [review]
Great work!

**Pull request merged**